File: library/src/androidTest/java/com/raizlabs/android/dbflow/test/structure/TestEmptyModel.java
package com.raizlabs.android.dbflow.test.structure;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;
import com.raizlabs.android.dbflow.test.TestDatabase;

/**
 * Description: Tests to ensure the allFields flag works as expected.
 */
@Table(databaseName = TestDatabase.NAME, allFields = true)
public class TestEmptyModel extends BaseModel {

    @Column(columnType = Column.PRIMARY_KEY)
    String name;

    int count;

    boolean truth;

    static int COUNT;

    final String finalName = "";

    private int hidden;
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/structure/TestEmptyModel.kt
package com.raizlabs.android.dbflow.structure

import com.raizlabs.android.dbflow.annotation.Column
import com.raizlabs.android.dbflow.annotation.ColumnIgnore
import com.raizlabs.android.dbflow.annotation.PrimaryKey
import com.raizlabs.android.dbflow.annotation.Table
import com.raizlabs.android.dbflow.structure.BaseModel
import com.raizlabs.android.dbflow.TestDatabase

/**
 * Description: Tests to ensure the allFields flag works as expected.
 */
@Table(database = TestDatabase::class, allFields = true)
class TestEmptyModel : BaseModel() {

    @Column
    @PrimaryKey
    var name: String? = null

    var count: Int = 0

    @JvmSynthetic
    var truth: Boolean = false

    internal val finalName = ""

    @ColumnIgnore
    private val hidden: Int = 0

    companion object {

        var COUNT: Int = 0
    }
}
